draft-ietf-quic-manageability.md

```diff
@@ -286,10 +286,17 @@ datagram as shown in {{fig-server-hello}} typically containing three packets:
 an Initial packet with the Server Hello, a Handshake packet with the rest of
 the server's side of the TLS handshake, and initial 1-RTT data, if present.
 
+The Client Hello datagram, Server Hello datagram, and Initial Completion datagram,
+which all contain a QUIC Initial Packet, must be at least 1200 octets long. This
```

Comment:
It is possible that the client will not include an Initial packet in its second flight.  Perhaps more importantly, if the client sends a Handshake in the same datagram, it is not required to pad that datagram.

Reply:
Did we make the Initial ACK formally optional, or are clients just omitting it as an optimization?

Sec 14.1 says "A client MUST expand the payload of all UDP datagrams carrying Initial packets to at least the smallest allowed maximum datagram size of 1200 bytes by adding PADDING frames to the Initial packet or by coalescing the Initial packet" without any qualification about Handshake.

My reading of the text is that it has to be 1200B, whether via PADDING or coalescing up to that point. I understand that there is little point to requiring that padding.

Reply:
Yes, if the client sends an Initial packet, it needs to be padded.  That's pretty clear.  My point is that it might not send it and so not cause that rule to apply.

Let's say that the client does send an Initial ACK before it discards keys.  That is lost.  It doesn't retransmit it because the keys are gone.  The observer won't see that; they only see the Handshake+ packet that is sent later.

And if the client does discard keys too soon, the protocol doesn't stop working and no rules have been broken.

I'm just asking that you be a little more careful and maybe say that a handshake will look like this *in most cases*.  There are too many ways this could be wrong and you used "must".

Reply:
OK, rewrote it, PTAL